x-pack/plugins/security_solution/server/lib/detection_engine/rule_types/factories/utils/generate_building_block_ids.ts

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ALERT_RULE_UUID } from '@kbn/rule-data-utils';
+import { createHash } from 'crypto';
+import { Ancestor } from '../../../signals/types';
+import { ALERT_ANCESTORS } from '../../field_maps/field_names';
+import { RACAlert } from '../../types';
+
+/**
+ * Generates unique doc ids for each building block signal within a sequence. The id of each building block
+ * depends on the parents of every building block, so that a signal which appears in multiple different sequences
+ * (e.g. if multiple rules build sequences that share a common event/signal) will get a unique id per sequence.
+ * @param buildingBlocks The full list of building blocks in the sequence.
+ */
+export const generateBuildingBlockIds = (buildingBlocks: RACAlert[]): string[] => {
```

Comment:
it looks like there are some similarities with `generateAlertId` function in `build_alert.ts`, do you think it makes sense to consolidate the two?